api/types.proto

```diff
@@ -713,3 +713,33 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
+	NODE = 1 [(gogoproto.enumvalue_customname) = "NodeSecret"];
+}
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretDataID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_data_id = 2 [(gogoproto.customname) = "SecretDataID"];
```

Comment:
Technically if the secret has `< 256` bits of information, this would give an attacker a better venue for discovering what the secret is.

Imagine that the secret is a 4 digit PIN. Having the `sha256(PIN)` is still trivially reversible.

Regardless of this, the main issue with using digest is that it doesn't allow us to point to different `SecretDatas` that have the same content (but potentially different annotations).